cylc/flow/data_store_mgr.py

```diff
@@ -318,6 +350,47 @@ def runtime_from_config(rtconfig):
     )
 
 
+def runtime_from_partial(rtconfig, runtimeold=None):
+    """Populate runtime object from partial/full config.
+
+    Potentially slower with all the setattr calls, but no expected fields.
+    """
+    runtime = PbRuntime()
+    if runtimeold is not None:
+        runtime.CopyFrom(runtimeold)
+    for key, val in rtconfig.items():
+        if val is None or key not in RUNTIME_MAPPING:
+            continue
+        elif key in RUNTIME_LIST_JOINS:
+            setattr(runtime, RUNTIME_MAPPING[key], listjoin(val))
+        elif key in RUNTIME_JSON_DUMPS:
+            setattr(
+                runtime,
+                RUNTIME_MAPPING[key],
+                json.dumps(
+                    [
+                        {'key': k, 'value': v}
+                        for k, v in val.items()
+                    ]
+                )
+            )
+        elif key in RUNTIME_TRY_ITEMS:
+            try:
+                setattr(
+                    runtime,
+                    RUNTIME_MAPPING[key],
+                    val[RUNTIME_TRY_ITEMS[key]]
+                )
+            except (KeyError, TypeError):
+                with suppress(KeyError, TypeError):
+                    setattr(runtime, RUNTIME_MAPPING[key], val)
```

Comment:
I'm not sure of the need for this, seems overly-cautious at the cost of readability to me 
```suggestion
        elif key == 'platform':
            setattr(runtime, RUNTIME_MAPPING[key], val['name'])
```

Reply:
That won't work, there are scenarios where `name` doesn't exist in `platform` and where `platform` doesn't exist..
The `runtime_from_config` has :
```
    try:
        platform = rtconfig['platform']['name']
    except (KeyError, TypeError):
        platform = rtconfig['platform']
  ```
  
 However this is passed in as an kwarg to `Runtime` .. Which can handle `None`s

Reply:
I've simplified this now.. Sometimes `val` is a dict that doesn't contain `name` so ignore the exception, otherwise `val` is a string so can use the `else` (at the end)